### ISSUE TITLE
Microsoft.ReactNative.vcxproj sets Link.AdditionalOptions without %(AdditionalOptions) discarding config from ExperimentalFeatures.props

### ISSUE THREAD

**Issue by nichamp** [microsoft/react-native-windows#9259]
### Problem Description

Microsoft.ReactNative.vcxproj sets Link.AdditionalOptions without %(AdditionalOptions) discarding config from ExperimentalFeatures.props such as setting "/DEBUGTYPE:CV,FIXUP" to allow Microsoft.ReactNative.dll to pass CfgChk as part of [PkgESSecComp](https://www.osgwiki.com/wiki/Package_ES_Security_and_Compliance) as is required for many Microsoft-internal pipelines. This appears to be a regression from #8823 (we have this issue with 0.66.3 but had no issues with 0.64.x).

### Steps To Reproduce

(1) Add something like the following to ExperimentalFeatures.props

```
  <ItemDefinitionGroup>
    <Link Condition="'$(Configuration)'=='Release'">
      <AdditionalOptions>/DEBUGTYPE:CV,FIXUP %(AdditionalOptions)</AdditionalOptions>
    </Link>
  </ItemDefinitionGroup>
  ```
  
(2) Set VS build output to detailed and observe that the additional flags are discarded

(3) Run PkgESSecComp@10 and observe it fails with something like ` BB0042: There is no fixup debug information in the image 'D:\a\_work\3\b\Native\Microsoft.ReactNative.dll'`

### Expected Results

The offending line in Microsoft.ReactNative.vcxproj should be `<AdditionalOptions>/IGNORE:4199 %(AdditionalOptions)</AdditionalOptions>`

### CLI version

N/A

### Environment

```markdown
RNW 0.66.3
```

### Target Platform Version

10.0.19041

### Visual Studio Version

Visual Studio 2019

**Comment by NickGerleman** [comment]
I know folks added instructions to do this into the RNW docs, but please be careful about adding MSBuild props to `ExperimentalFeatures.props` that are not already present and documented. I added the property sheet to allow customizing a couple of predefined properties that we know are safe, but folks discovered they could use it as a backdoor to accomplish other goals.

In the future we will be distributing projects as precompiled binaries, and it will not be possible to add arbitrary customizations to the build. Anything adding build-time props to `ExperimentalFeatures.props` will fail.

cc @asklar @jonthysell, since this issue seems to be popping up again.

**Comment by NickGerleman** [comment]
With all that said, I can't think of a reason not to add the specific flags used by PkgEs compliance tools to our official configuration. That would also add it to our officially distributed binaries.

**Comment by NickGerleman** [comment]
https://github.com/microsoft/react-native-windows/pull/9267 enables fixup debug information in our official logic

**Comment by nichamp** [comment]
> I know folks added instructions to do this into the RNW docs, but please be careful about adding MSBuild props to `ExperimentalFeatures.props` that are not already present and documented. I added the property sheet to allow customizing a couple of predefined properties that we know are safe, but folks discovered they could use it as a backdoor to inject arbitrary props.
> 
> In the future we will be distributing projects as precompiled binaries, and it will not be possible to add arbitrary customizations to the build. Anything adding build-time props to `ExperimentalFeatures.props` will fail.
> 
> cc @asklar @jonthysell, since this issue seems to be popping up again. I am very tempted to restrict `ExperimentalFeatures.props` since it seems to keep on getting used in ways that will regress once we are on binaries. We should not tell folks in documentation to use it for purposes that will not work in the future.

Photon will need to use ExperimentalFeatures.props and build from source as there are a number of other custom configurations that are required for our scenarios beyond compiler flags for compliance so consuming the NuGet packages would not be likely.


> With all that said, we should explore adding the specific flags used by PkgEs compliance tools to our official configuration. That would also add it to our officially distributed binaries. I can't think of a compelling reason not to add this extra information to PDBs.

That is true, but that alone is not the only flag required for compliance and that one merely allows CfgChk to proceed. Others are required for CheckCFlags2 too.

**Comment by NickGerleman** [comment]
The MSBuild properties internal to our projects are not a public surface, and will continue to change.

@nichamp let me know if you'd like to chat later this week about how we might be able to meet Photon's goals. I have a feeling we can do a lot with public surfaces.

Otherwise a common tactic in the RN ecosystem is to maintain a source-fork, with any sort of changes to build logic there.

**Comment by nichamp** [comment]
I don't think most of the items in the link I provided above would have such existing "public surface". For the purpose of this issue, I am just asking that msbuild variables follow the usual convention of checking if something is already set before assigning, or for lists aggregating accordingly.

**Comment by NickGerleman** [comment]
I did some cleanup for that specific change here https://github.com/microsoft/react-native-windows/pull/9266/files

Just want to make sure there's clarity that, in the general sense, changes to the MSBuild properties internal to our projects are expected, and will  happen version to version.

**Comment by nichamp** [comment]
Yeah, understood. I wasn't expecting any warranties given custom msbuild variables are not an API surface - just that we can quickly address the issues when they come up. Though for more standard ones defined for typical vcxproj (e.g. compiler and UWP props) I would hope we can keep those configurable.

**Comment by NickGerleman** [comment]
That list, is I think what we should try to make public.

Right now, when you generate a new app, you get an "ExperimentalFeatures.props" file with a list of default settings and their descriptions.

We could expose a similar file, for specifics like Windows SDK, Linker settings, WinUI version, etc.

That helps create a contact, of the properties you can customize for doing custom source builds.
